user-console/src/main/java/org/pentaho/mantle/client/ui/BurgerMenuBar.java

```diff
@@ -0,0 +1,320 @@
+    if ( menuPopup != null ) {
+      menuPopup.hide();
+    }
+  }
+
+  protected BurgerMenuBar getRootMenu() {
+    BurgerMenuBar parentMenu = getMenuBarParent();
+    if ( parentMenu != null ) {
+      return parentMenu.getRootMenu();
+    }
+
+    return this;
+  }
+
+  protected PopupPanel getRootPopup() {
+    Widget parent = getRootMenu().getParent();
+    while ( parent != null ) {
+      if ( parent instanceof PopupPanel ) {
+        return (BurgerMenuPopup) parent;
+      }
+
+      parent = parent.getParent();
+    }
+
+    return null;
+  }
+
+  // Local version of private super.findItem( Element ).
+  protected MenuItem findItemLocal( com.google.gwt.dom.client.Element hItem ) {
+    for ( MenuItem item : getItems() ) {
+      if ( item.getElement().isOrHasChild( hItem ) ) {
+        return item;
+      }
+    }
+
+    return null;
+  }
+
+  // Local version of private super.eatEvent( Event ).
+  protected void eatEventLocal( Event event ) {
+    event.stopPropagation();
+    event.preventDefault();
+  }
+
+  // Local version of private super.close( boolean ).
+  protected void closeLocal( boolean focus ) {
+    BurgerMenuBar parentMenuBar = getMenuBarParent();
+    if ( parentMenuBar != null ) {
+      PopupPanel popupPanel = parentMenuBar.getMenuBarPopup();
+      if ( popupPanel != null ) {
+        popupPanel.hide( !focus );
+      }
+
+      if ( focus ) {
+        parentMenuBar.focus();
+      }
+    }
+  }
+
+  // Local version of private super.selectFirstItemIfNoneSelected().
+  private boolean selectFirstItemIfNoneSelectedLocal() {
+    if ( getSelectedItem() == null ) {
+      for ( MenuItem nextItem : getItems() ) {
+        if ( nextItem.isEnabled() ) {
+          selectItem( nextItem );
+          break;
+        }
+      }
+
+      return true;
+    }
+
+    return false;
+  }
+
+  // Local, adapted version of private super.doItemAction(MenuItem, boolean, boolean).
+  // Overrides the command execution of a menu item closes the whole menu (unlike closeAllParents()).
+  // Additionally, the special back menu items do not cause the menu to close.
+  protected void doItemActionLocal( MenuItem item, boolean fireCommand, boolean focus ) {
+    if ( !item.isEnabled() ) {
+      return;
+    }
+
+    if ( fireCommand && item.getScheduledCommand() != null ) {
+      FocusImpl.getFocusImplForPanel().blur( getElement() );
+
+      if ( !( item instanceof BurgerBackMenuItem ) ) {
+        closeRoot();
+      }
+
+      Scheduler.ScheduledCommand cmd = item.getScheduledCommand();
+      Scheduler.get().scheduleFinally( cmd::execute );
+    } else {
+      doItemActionLocalSuper( item, fireCommand, focus );
+    }
+
+    if ( item.getScheduledCommand() == null && item.getSubMenu() != null ) {
+      item.getSubMenu().focus();
+    }
+  }
+
+  // Local version of private super.doItemAction(MenuItem, boolean, boolean).
+  private native void doItemActionLocalSuper( MenuItem item, boolean fireCommand, boolean focus ) /*-{
```

Comment:
Since this is the same design pattern, accessing non-public/protected methods via JSNI, shouldn't this be moved to repo pentaho-commons-gwt-modules with the class MenuBarUtils for consistency?
 https://github.com/pentaho/pentaho-commons-gwt-modules/blob/wcag/widgets/src/main/java/org/pentaho/gwt/widgets/client/utils/MenuBarUtils.java

Reply:
addressed with PR https://github.com/pentaho/pentaho-commons-gwt-modules/pull/917